packages/shared/src/lib/cloud/patches/events.ts

```diff
@@ -0,0 +1,147 @@
+import { CloudPatch, type ApiPatch } from '$lib/cloud/types';
+import { derived, type Readable } from 'svelte/store';
+import type { HttpClient } from '$lib/httpClient';
+
+interface ApiChatUser {
+	id: number;
+	login: string;
+	name: string;
+	email: string;
+	avatar_url?: string;
+}
+
+class CloudChatUser {
+	id: number;
+	login: string;
+	name: string;
+	email: string;
+	avatarUrl?: string;
+
+	constructor(apiChatUser: ApiChatUser) {
+		this.id = apiChatUser.id;
+		this.login = apiChatUser.login;
+		this.name = apiChatUser.name;
+		this.email = apiChatUser.email;
+		this.avatarUrl = apiChatUser.avatar_url;
+	}
+}
+
+interface DiffPatchEntry {
+	type: 'context' | 'added' | 'removed';
+	left?: number;
+	right?: number;
+	line: string;
+}
+
+interface ApiChatMessage {
+	uuid: string;
+	user: ApiChatUser;
+	outdated: boolean;
+	issue: boolean;
+	resolved: boolean;
+	thread_id?: string;
+	comment: unknown;
+
+	// data needed if commenting on part of a patch
+	diff_sha?: string;
+	range?: string;
+	diff_patch_array?: DiffPatchEntry[];
+
+	created_at: string;
+	updated_at: string;
+}
+
+class CloudChatMessage {
+	uuid: string;
+	user: CloudChatUser;
+	outdated: boolean;
+	issue: boolean;
+	resolved: boolean;
+	threadId?: string;
+	comment: unknown;
+
+	// data needed if commenting on part of a patch
+	diffSha?: string;
+	range?: string;
+	diffPatchArray?: DiffPatchEntry[];
+
+	createdAt: Date;
+	updatedAt: Date;
+
+	constructor(apiChatMessage: ApiChatMessage) {
+		this.uuid = apiChatMessage.uuid;
+		this.user = new CloudChatUser(apiChatMessage.user);
+		this.outdated = apiChatMessage.outdated;
+		this.issue = apiChatMessage.issue;
+		this.resolved = apiChatMessage.resolved;
+		this.threadId = apiChatMessage.thread_id;
+		this.comment = apiChatMessage.comment;
+		this.diffSha = apiChatMessage.diff_sha;
+		this.range = apiChatMessage.range;
+		this.diffPatchArray = apiChatMessage.diff_patch_array;
+		this.createdAt = new Date(apiChatMessage.created_at);
+		this.updatedAt = new Date(apiChatMessage.updated_at);
```

Comment:
Could this be replaced with `Object.assign(this, apiChatMessage)`?

Reply:
no, because there are casing changes

Reply:
and type changes